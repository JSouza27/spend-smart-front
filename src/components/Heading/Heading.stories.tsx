import { Story, ComponentMeta } from '@storybook/react';
import { ThemeProvider } from 'styled-components';

import Heading, { HeadingProps } from '.';
import theme from '../../styles/theme';

export default {
  title: 'Heading',
  component: Heading,
  args: {
    color: 'neutral_900',
    fontWeight: 600,
    level: 1,
    lineHeight: '4rem',
    size: 'xxxlarge'
  }
} as ComponentMeta<typeof Heading>;

export const Default: Story<HeadingProps> = (args) => (
  <ThemeProvider theme={theme}>
    <Heading {...args}>Minhas Transações</Heading>
  </ThemeProvider>
);
