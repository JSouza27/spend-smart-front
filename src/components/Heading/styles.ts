import styled, { css } from 'styled-components';
import { HeadingProps } from '.';

export const Heading = styled.h1.attrs<HeadingProps>(({ level }) => ({
  as: `h${level}`
}))<HeadingProps>`
  ${({
    color = 'neutral_900',
    size = 'xxxlarge',
    fontWeight = 600,
    lineHeight = '4rem'
  }) => css`
    color: ${color};
    font-size: ${size};
    font-weight: ${fontWeight};
    line-height: ${lineHeight};
  `}
`;
