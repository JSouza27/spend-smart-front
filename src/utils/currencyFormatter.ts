export class CurrencyFormatter {
  private static language = 'pt-BR';
  private static currency = 'BRL';

  static formatter(num: number): string {
    return (num ?? 0).toLocaleString(this.language, {
      currency: this.currency
    });
  }

  static unformat(num: string): number {
    const nums = num.replace(/[^0-9]+/g, '');
    const i = nums.length - 2;

    return Number(`${nums.substring(0, i)}.${nums.substring(i)}`);
  }
}
