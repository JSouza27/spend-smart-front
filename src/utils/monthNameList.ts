export const monthNameList = [
  'janeiro',
  'fevereiro',
  'mar√ßo',
  'abril',
  'maio',
  'junho',
  'julho',
  'agosto',
  'setembro',
  'outubro',
  'novembro',
  'dezembro'
];
