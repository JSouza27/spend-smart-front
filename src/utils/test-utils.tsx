import { render } from '@testing-library/react';
import React from 'react';
import { ThemeProvider } from 'styled-components';

import theme from '../styles/theme';
import { FormProvider, useForm } from 'react-hook-form';

const AllTheProviders = ({ children }: any) => {
  const methods = useForm();

  return (
    <ThemeProvider theme={theme}>
      <FormProvider {...methods}>{children}</FormProvider>
    </ThemeProvider>
  );
};

const customRender = (
  ui: React.ReactElement<any, string | React.JSXElementConstructor<any>>,
  options?: any
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export { customRender as render };
