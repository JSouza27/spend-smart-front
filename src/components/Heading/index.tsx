import React, { HTMLAttributes } from 'react';

import * as S from './styles';
import theme from '../../styles/theme';

export type HeadingProps = {
  color?: keyof typeof theme.colors;
  size?: keyof typeof theme.font.sizes;
  fontWeight?: 400 | 600;
  lineHeight?: string | number;
  level?: 1 | 2 | 3 | 4 | 5 | 6;
  children: React.ReactNode;
} & HTMLAttributes<HTMLHeadingElement>;

export default function Heading({ children, ...props }: HeadingProps) {
  return <S.Heading {...props}>{children}</S.Heading>;
}
