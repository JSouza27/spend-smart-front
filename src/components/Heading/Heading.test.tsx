import { render, screen } from 'utils/test-utils';

import Heading from '.';

describe('<Heading />', () => {
  it('should render the heading', () => {
    const { container } = render(<Heading level={1}>Heading</Heading>);

    const heading = screen.getByRole('heading', {
      name: /heading/i
    });

    expect(heading).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
