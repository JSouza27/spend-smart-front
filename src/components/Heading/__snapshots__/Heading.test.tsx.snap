// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> should render the heading 1`] = `
.c0 {
  color: neutral_900;
  font-size: xxxlarge;
  font-weight: 600;
  line-height: 4rem;
}

<h1
  class="c0"
>
  Heading
</h1>
`;
